### PR TITLE
dev/core#3783 convert Recent Items providers into an option group

### PR DESCRIPTION
**Pull request by herbdool**
Overview
----------------------------------------
Fixes https://lab.civicrm.org/dev/core/-/issues/3783

Before
----------------------------------------
Recent Items providers is a hard-coded list.

After
----------------------------------------
Adds an option group and uses that instead.

I'm not quite sure if I've got the right place to put the upgrade query. I haven't tried it for core yet.

### CODE REVIEW

File: CRM/Upgrade/Incremental/sql/5.53.alpha1.mysql.tpl [hunk 1]
```diff
@@ -1 +1,22 @@
...
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Activities{/ts}'{/localize}, 'Activity', 'Activities', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Notes{/ts}'{/localize}, 'Note', 'Notes', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Groups{/ts}'{/localize}, 'Group', 'Groups', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Cases{/ts}'{/localize}, 'Case', 'Cases', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Contributions{/ts}'{/localize}, 'Contribution', 'Contributions', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Participants{/ts}'{/localize}, 'Participant', 'Participants', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
+    (@option_group_id_recent, {localize}'{ts escape="sql"}Grants{/ts}'{/localize}, 'Grant', 'Grants', NULL, NULL, 0, 1, '', 0, 0, 1, NULL, NULL),
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/sql/5.53.alpha1.mysql.tpl:18]: This one shouldn't be here. I think it best to put it in a `.mgd.php` file in the CiviGrant extension.

**Comment by colemanw** [comment]
Normally we try to write upgrade code in a way that's safe to be run multiple times.
The way this upgrader is written, it's going to crash if it's already been run once.
I think the way the upgrade code is written is actually the right way to write *install* sql, and it should be moved to the installer.
For the upgrader, I would do it in php rather than a sql file, and take advantage of the failsafe functions `\CRM_Core_BAO_OptionGroup::ensureOptionGroupExists` and `\CRM_Core_BAO_OptionValue::ensureOptionValueExists`

File: CRM/Upgrade/Incremental/sql/5.53.alpha1.mysql.tpl (hunk 1 above)
Comment (herbdool) [reply CRM/Upgrade/Incremental/sql/5.53.alpha1.mysql.tpl]: Done

**Comment by herbdool** [comment]
I've updated it. I figured eventually it would be in php, but as I was looking around it seemed like it would temporarily in `sql/*.tpl` until closer to the release. Anyway, I guess I was wrong.

**Comment by colemanw** [comment]
@herbdool this looks good. The sql you wrote is still needed though, it just needs to be placed in `civicrm_data.tpl` for the sake of new installs.

**Comment by herbdool** [comment]
@colemanw I added it to `civicrm_data.tpl` too. All tests pass

**Comment by colemanw** [comment]
I tested this and added one additional safeguard. Everything else looks fine and works as expected.

**Pull request merged**